aspnetcore/fundamentals/configuration/options.md

```diff
@@ -22,259 +22,101 @@ Options also provide a mechanism to validate configuration data. For more inform
-
-[!code-csharp[](options/samples/3.x/OptionsSample/Models/MyOptions.cs?name=snippet1)]
-
-The `MyOptions` class is added to the service container with <xref:Microsoft.Extensions.DependencyInjection.OptionsConfigurationServiceCollectionExtensions.Configure*> and bound to configuration:
-
-[!code-csharp[](options/samples/3.x/OptionsSample/Startup.cs?name=snippet_Example1)]
-
-The following page model uses [constructor dependency injection](xref:mvc/controllers/dependency-injection) with <xref:Microsoft.Extensions.Options.IOptionsMonitor%601> to access the settings (*Pages/Index.cshtml.cs*):
+<a name="optpat"></a>
 
-[!code-csharp[](options/samples/3.x/OptionsSample/Pages/Index.cshtml.cs?range=9)]
-
-[!code-csharp[](options/samples/3.x/OptionsSample/Pages/Index.cshtml.cs?name=snippet2&highlight=2,8)]
-
-[!code-csharp[](options/samples/3.x/OptionsSample/Pages/Index.cshtml.cs?name=snippet_Example1)]
-
-The sample's *appsettings.json* file specifies values for `option1` and `option2`:
-
-[!code-json[](options/samples/3.x/OptionsSample/appsettings.json?highlight=2-3)]
-
-When the app is run, the page model's `OnGet` method returns a string showing the option class values:
-
-```html
-option1 = value1_from_json, option2 = -1
-```
-
-> [!NOTE]
-> When using a custom <xref:System.Configuration.ConfigurationBuilder> to load options configuration from a settings file, confirm that the base path is set correctly:
->
-> ```csharp
-> var configBuilder = new ConfigurationBuilder()
->    .SetBasePath(Directory.GetCurrentDirectory())
->    .AddJsonFile("appsettings.json", optional: true);
-> var config = configBuilder.Build();
->
-> services.Configure<MyOptions>(config);
-> ```
->
-> Explicitly setting the base path isn't required when loading options configuration from the settings file via <xref:Microsoft.AspNetCore.WebHost.CreateDefaultBuilder*>.
-
-## Configure simple options with a delegate
-
-Configuring simple options with a delegate is demonstrated as Example 2 in the sample app.
-
-Use a delegate to set options values. The sample app uses the `MyOptionsWithDelegateConfig` class (*Models/MyOptionsWithDelegateConfig.cs*):
-
-[!code-csharp[](options/samples/3.x/OptionsSample/Models/MyOptionsWithDelegateConfig.cs?name=snippet1)]
-
-In the following code, a second <xref:Microsoft.Extensions.Options.IConfigureOptions%601> service is added to the service container. It uses a delegate to configure the binding with `MyOptionsWithDelegateConfig`:
+## Bind hierarchical configuration
 
-[!code-csharp[](options/samples/3.x/OptionsSample/Startup.cs?name=snippet_Example2)]
+[!INCLUDE[](~/includes/bind.md)]
```

Comment:
I'm not sure how well this works as an include. The first line in the include links to "options pattern", which is the current topic when viewed here. Could the content just live in this topic and be linked over from the configuration topic, rather than it being embedded in both?

Reply:
It's not great as an include and the link is a bit klunky, but both topics need this information. We can think about moving it out in another PR.